pkg/healthcheck/healthcheck_test.go

```diff
@@ -1774,6 +1781,215 @@ data:
+		},
+
+		{
+			checkDescription:   "fails when cert dns is wrong",
+			certificateDNSName: "wrong.linkerd.cluster.local",
+			expectedErr:        errors.New("invalid credentials: x509: certificate is valid for wrong.linkerd.cluster.local, not identity.linkerd.cluster.local"),
+			fileNameToSave:     "wrong-domain",
+		},
+
+		{
+			checkDescription: "fails when cert is not valid yet",
+			lifespan: &lifeSpan{
+				starts: time.Date(2100, 1, 1, 1, 1, 1, 1, time.UTC),
+				ends:   time.Date(2101, 1, 1, 1, 1, 1, 1, time.UTC),
+			},
+			expectedErr:    errors.New("certificate not valid before: 2100-01-01T01:00:51Z"),
+			fileNameToSave: "not-valid-yet",
+		},
+
+		{
+			checkDescription: "fails when cert is expired",
+			lifespan: &lifeSpan{
+				starts: time.Date(1989, 1, 1, 1, 1, 1, 1, time.UTC),
+				ends:   time.Date(1990, 1, 1, 1, 1, 1, 1, time.UTC),
+			},
+			expectedErr:    errors.New("certificate not valid anymore. Expired at: 1990-01-01T01:01:11Z"),
+			fileNameToSave: "expired",
+		},
+
+		{
+			checkDescription: "warns if certificate expires to soon",
+			lifespan: &lifeSpan{
+				starts: time.Date(1989, 1, 1, 1, 1, 1, 1, time.Local),
+				ends:   time.Now().AddDate(0, 0, 1),
+			},
+			expectedWarning: errors.New("certificate is expiring sooner than 60 days from now"),
+		},
+	}
+
+	for id, testCase := range testCases {
+		testCase := testCase
+
+		if testCase.certificateDNSName == "" {
+			testCase.certificateDNSName = "identity.linkerd.cluster.local"
+		}
+		if testCase.lifespan == nil {
+			testCase.lifespan = &lifeSpan{
+				starts: time.Now().AddDate(-1, 0, 0),
+				ends:   time.Now().AddDate(1, 0, 0),
+			}
+		}
+
+		if testCase.schemeInConfig == "" && testCase.tlsSecretScheme == "" {
+			testCase.schemeInConfig = k8s.IdentityIssuerSchemeLinkerd
+			testCase.tlsSecretScheme = k8s.IdentityIssuerSchemeLinkerd
+		}
+
+		t.Run(fmt.Sprintf("%d", id), func(t *testing.T) {
+			hc := NewHealthChecker([]CategoryID{}, &Options{})
+			hc.ControlPlaneNamespace = "linkerd"
+
+			issuerData := createIssuerData(testCase.certificateDNSName, testCase.lifespan.starts, testCase.lifespan.ends)
+			config := getFakeConfig(testCase.tlsSecretScheme, testCase.schemeInConfig, issuerData)
+
+			/*			o.heartbeatSchedule = fakeHeartbeatSchedule
+						o.identityOptions.crtPEMFile = filepath.Join("testdata", "crt.pem")
+						o.identityOptions.keyPEMFile = filepath.Join("testdata", "key.pem")
+						o.identityOptions.trustPEMFile = filepath.Join("testdata", "trust-anchors.pem")
+			*/
+
+			if testCase.fileNameToSave != "" {
+				crtFile := path.Join("/Users/zaharidichev/work/linkerd2/testCerts/testData/", testCase.fileNameToSave+"-crt.pem")
+				keyFile := path.Join("/Users/zaharidichev/work/linkerd2/testCerts/testData/", testCase.fileNameToSave+"-key.pem")
+				anchors := path.Join("/Users/zaharidichev/work/linkerd2/testCerts/testData/", testCase.fileNameToSave+"-trust-anchors.pem")
+
+				ioutil.WriteFile(crtFile, []byte(issuerData.IssuerCrt), 0777)
```

Comment:
actually, I don't even see where these files are used.  is this needed?

Reply:
oh, yes I used the test to generate the test files used in the install cmd tests. Must have left that there by mistake.